projects/ngx-prx-styleguide/src/lib/tz-datepicker/tz-data.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { TzDatepickerModule } from './tz-datepicker.module';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: TzDatepickerModule
```

Comment:
We get a circular dependency warning during compilation (as this causes the `TzDataService` to depend on `TzDatepickerModule`)

It seems like the general advice to solve this problem is "don't do it" (always use `providedIn: 'root'`), but I'm not sure how that would play out when the root module is in our consumer.

Reply:
From the Angular providers guide:
> In an eagerly loaded app, the root application injector makes all of the providers in all of the modules available throughout the app.

From that, I'd guess that adding `TzDatepickerModule` to the consumer's imports would be enough to pull in this provider too, since it's in the module's providers. I'll just annotate this as  injectable without any configuration.